temporalcli/client.go

```diff
@@ -92,6 +101,25 @@ func (c *ClientOptions) tlsConfig() (*tls.Config, error) {
 	return conf, nil
 }
 
+func fixedHeaderOverrideInterceptor(
+	ctx context.Context,
+	method string, req, reply any,
+	cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption,
+) error {
+	// The SDK sets some values on the outgoing metadata that we can't override
+	// via normal headers, so we have to replace directly on the metadata
+	md, _ := metadata.FromOutgoingContext(ctx)
+	if md == nil {
+		md = metadata.MD{}
+	}
+	md.Set("client-name", "temporal-cli")
+	md.Set("client-version", Version)
+	md.Set("supported-server-versions", ">=1.0.0 <2.0.0")
+	md.Set("caller-type", "operator")
```

Comment:
What is this one for?

Reply:
It gives calls made by the CLI higher priority over application code.
This helps troubleshoot issues when you're getting rate limited for example.